src/site/shared/containers/HistoryBox/index.tsx

```diff
@@ -69,21 +100,48 @@ export const HistoryBox = ({ info }: Props) => {
         state => ({ ...state.itemNav })
     );
     const dispatch = useSharedDispatch();
-
+    
     const { undoHistory, redoHistory } = useHistory(info);
 
-    return (
-        <div className={`historybox ${isOpen ? "" : "historybox__move"} ${isHistoryBoxOpen ? "" : "hide"}`}>
-            <div>
-                <span>History</span>
-                <span onClick={() => dispatch(CloseHistoryBox())}>×</span>
+    var redoHasMembers = false;
+    if(redoHistory.length > 0) {
+        return (
+            <div className={`historybox ${isOpen ? "" : "historybox__move"} ${isHistoryBoxOpen ? "" : "hide"}`}>
+                <div>
+                    <span>History</span>
+                    <span onClick={() => dispatch(CloseHistoryBox())}>×</span>
+                </div>
+                <div>
+                    {[...redoHistory].reverse().map((a, i) =>
+                        <HistoryEntry key={`history-box-dashedentry-${i}`} a={a} isRedo></HistoryEntry>
+                    )}
+                    <div style={{textAlign: 'center', fontWeight: 'bold'}}> Redo </div>
+                    <div className={"historybox__separator"} > </div>
+                    <div style={{textAlign: 'center', fontWeight: 'bold'}}> Undo </div>
+                    {[...undoHistory].reverse().map((a, i) =>
+                        <HistoryEntry key={`history-box-entry-${i}`} a={a} isRedo={false}></HistoryEntry>
+                    )}
+                </div>
             </div>
-            <div>
-                {[...undoHistory].reverse().map((a, i) =>
-                    <HistoryEntry key={`history-box-entry-${i}`} a={a}></HistoryEntry>
-                )}
+
+        );
+    }
+    else {
+        return (
+            <div className={`historybox ${isOpen ? "" : "historybox__move"} ${isHistoryBoxOpen ? "" : "hide"}`}>
+                <div>
+                    <span>History</span>
+                    <span onClick={() => dispatch(CloseHistoryBox())}>×</span>
+                </div>
+                <div>
+                    <div style={{textAlign: 'center', fontWeight: 'bold'}}> Undo </div>
+                    {[...undoHistory].reverse().map((a, i) =>
+                        <HistoryEntry key={`history-box-entry-${i}`} a={a} isRedo={false}></HistoryEntry>
+                    )}
+                    
+                </div>
```

Comment:
Similar to the large if/else statement above, this can probably be compressed. Inside of the html sections, you can use the `{}` braces to enter back into the javascript. Bascially, you would want something like this:
```ts
{ if (redoHistory.length > 0) {<>
    {[...redoHistory].reverse().map((a, i) =>
        <HistoryEntry key={`history-box-dashedentry-${i}`} a={a} isRedo></HistoryEntry>
    )}
    <div style={{textAlign: 'center', fontWeight: 'bold'}}> Redo </div>
    <div className={"historybox__separator"} > </div>
</>}
}
```
I may have the syntax a little off there, but that reduce the amount of code reused between the if and the else, since it looks like everything else is the same.